client/state/login/actions.js

```diff
@@ -202,8 +215,16 @@ export const loginSocialUser = ( service, token ) => dispatch => {
 			dispatch( { type: SOCIAL_LOGIN_REQUEST_SUCCESS } );
 		} )
 		.catch( ( httpError ) => {
+			const errorKey = get( httpError, 'response.body.data.errors[0]' );
 			const error = getErrorFromHTTPError( httpError );
 
+			if ( errorKey === 'unknown_user' ) {
+				if ( ( typeof shouldCreateAccount === 'function' && shouldCreateAccount() ) ||
+					shouldCreateAccount ) {
```

Comment:
If `shouldCreateAccount` is always one type (only ever a function, currently), I think handling an additional type just makes the code a little harder to follow.